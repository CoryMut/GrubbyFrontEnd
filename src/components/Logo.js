import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Link, useLocation } from "react-router-dom";

const useStyles = makeStyles((theme) => ({
    logo: {
        cursor: "pointer",
        maxWidth: "230px",
        width: "100%",
        userSelect: "none",
        [theme.breakpoints.down("md")]: {
            width: "90%",
        },
    },
}));

const Logo = () => {
    const classes = useStyles();
    const location = useLocation();

    const handleClick = () => {
        if (location.pathname === "/") {
            window.scrollTo({ top: 0, behavior: "smooth" });
        }
    };

    return (
        <Link to="/" onClick={handleClick}>
            <img
                className={classes.logo}
                src="https://grubbythegrape.sfo2.cdn.digitaloceanspaces.com/assets/assets/logo/white_logo_crop.png"
                alt="Grubby the Grape logo"
            ></img>
        </Link>
    );
};

export default Logo;
