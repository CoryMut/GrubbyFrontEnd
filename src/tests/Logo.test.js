import React from "react";
import { shallow } from "enzyme";
import Logo from "../components/Logo";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
        pathname: "/",
    }),
}));

test("should test Logo component", () => {
    const wrapper = shallow(<Logo />);
    expect(wrapper).toMatchSnapshot();
});

test("it renders without crashing", () => {
    render(
        <MemoryRouter>
            <Logo />
        </MemoryRouter>
    );
});

test("should test Logo component", () => {
    const scrollToSpy = jest.fn();
    global.scrollTo = scrollToSpy;
    const wrapper = shallow(<Logo />);
    wrapper.find("Link").simulate("click");
    expect(location.pathname).toBe("/");
    expect(scrollToSpy).toHaveBeenCalled();
});
