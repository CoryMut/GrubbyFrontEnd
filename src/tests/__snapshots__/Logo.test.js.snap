// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Logo component 1`] = `
<Fragment>
  <Link
    onClick={[Function]}
    to="/"
  >
    <img
      alt="Grubby the Grape logo"
      className="makeStyles-logo-1"
      src="https://grubbythegrape.sfo2.cdn.digitaloceanspaces.com/assets/assets/logo/logo%28325x55%29.png"
    />
  </Link>
</Fragment>
`;
